src/pkcs15init/profile.c

```diff
@@ -2006,6 +2019,9 @@ build_argv(struct state *cur, const char *cmdname,
 		if (list == mac->value) {
 			return SC_ERROR_SYNTAX_ERROR;
 		}
+		if (check_macro_cycle(list, mac->value, cur)) {
```

Comment:
This change is quite hard to understand. It would be great if this would have some comment describing what is the issue we are trying to solve and how we are solving it.

My understanding is that this means there are some macros that recursively calls itself causing the infinite loop/recursion (maybe better word than cycle?) evaluation.